R/recipe.R

```diff
@@ -80,231 +81,47 @@ axe_env.step_arrange <- function(x, ...) {
-  x$terms <- purrr::map(x$terms, function(z) axe_env(z, ...))
-  x
-}
-
-#' A means to replace environments wrapped from the \code{step_factor2string} function.
-#'
-#' @rdname axe-recipe
-#' @export
-axe_env.step_factor2string <- function(x, ...) {
-  x$terms <- purrr::map(x$terms, function(z) axe_env(z, ...))
-  x
-}
-
 #' A means to replace environments wrapped from the \code{step_filter} function.
 #'
 #' @rdname axe-recipe
 #' @export
 axe_env.step_filter <- function(x, ...) {
-  x$inputs <- purrr::map(x$inputs, function(z) axe_env(z, ...))
-  x
+  axe_env.step_arrange(x, ...)
 }
 
-#' A means to replace environments wrapped from the \code{step_geodist} function.
-#'
-#' @rdname axe-recipe
-#' @export
-axe_env.step_geodist <- function(x, ...) {
-  x$lat <- purrr::map(x$lat, function(z) axe_env(z, ...))
-  x$lon <- purrr::map(x$lon, function(z) axe_env(z, ...))
-  x
-}
-
-#' A means to replace environments wrapped from the \code{step_holiday} function.
-#'
-#' @rdname axe-recipe
-#' @export
-axe_env.step_holiday <- function(x, ...) {
-  x$terms <- purrr::map(x$terms, function(z) axe_env(z, ...))
-  x
-}
-
-#' A means to replace environments wrapped from the \code{step_hyperbolic} function.
-#'
-#' @rdname axe-recipe
-#' @export
-axe_env.step_hyperbolic <- function(x, ...) {
-  x$terms <- purrr::map(x$terms, function(z) axe_env(z, ...))
-  x
-}
-
-#' A means to replace environments wrapped from the \code{step_ica} function.
-#'
-#' @rdname axe-recipe
-#' @export
-axe_env.step_ica <- function(x, ...) {
-  x$terms <- purrr::map(x$terms, function(z) axe_env(z, ...))
-  x
-}
-
-#' A means to replace environments wrapped from the \code{step_integer} function.
-#'
-#' @rdname axe-recipe
-#' @export
-axe_env.step_integer <- function(x, ...) {
-  x$terms <- purrr::map(x$terms, function(z) axe_env(z, ...))
-  x
-}
-
-#' A means to replace environments wrapped from the \code{step_interact} function.
+#' A means to replace environments wrapped from the \code{step_mutate} function.
 #'
 #' @rdname axe-recipe
 #' @export
-axe_env.step_interact <- function(x, ...) {
-  attr(x$terms, ".Environment") <- rlang::base_env()
-  x
+axe_env.step_mutate <- function(x, ...) {
+  axe_env.step_arrange(x, ...)
 }
 
-#' A means to replace environments wrapped from the \code{step_inverse} function.
+#' A means to replace environments wrapped from the \code{step_slice} function.
 #'
 #' @rdname axe-recipe
 #' @export
-axe_env.step_inverse <- function(x, ...) {
-  x$terms <- purrr::map(x$terms, function(z) axe_env(z, ...))
-  x
+axe_env.step_slice <- function(x, ...) {
+  axe_env.step_arrange(x, ...)
 }
 
-#' A means to replace environments wrapped from the \code{step_invlogit} function.
+#' A means to replace environments wrapped from the \code{step_impute_bag} function.
 #'
 #' @rdname axe-recipe
 #' @export
-axe_env.step_invlogit <- function(x, ...) {
+axe_env.step_impute_bag <- function(x, ...) {
   x$terms <- purrr::map(x$terms, function(z) axe_env(z, ...))
```

Comment:
Maybe replace this with `x <- NextMethod()` now? Same for `axe_env.step_impute_knn()` and `axe_env.step_ratio()`